conans/client/graph/graph_binaries.py

```diff
@@ -160,59 +161,65 @@ def _process_node(self, node, pref, build_mode):
             node.binary = BINARY_EDITABLE  # TODO: PREV?
             return
 
-        cache_latest_prev = self._cache.get_latest_package_reference(pref)
-
-        if not cache_latest_prev:  # This binary does NOT exist in the cache
-            try:
-                remote = self._get_package_from_remotes(node, pref)
-            except NotFoundException:
-                node.binary = BINARY_MISSING
-                node.prev = None
-                node.binary_remote = None
-            else:
-                node.binary = BINARY_DOWNLOAD
-                node.prev = pref.revision
-                node.pref_timestamp = pref.timestamp
-                node.binary_remote = remote
-        else:  # This binary already exists in the cache
+        # Obtain the cache_latest valid one, cleaning things if dirty
+        while True:
```

Comment:
This was previously wrong, as it was doing the potential cleaning, after checking the ``cache_latest_prev``, but never invalidating. Probably under-tested